src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3242,7 +3247,8 @@ def stop(self):
         # During LRMS termination, we call any existing shutdown hooks on the
         # launch methods.  We only call LM shutdown hooks *once*
         launch_methods = set() # set keeps entries unique
-        launch_methods.add(self._cfg['mpi_launch_method'])
+        if 'mpi_launch_method' in self._cfg:
+            launch_methods.add(self._cfg['mpi_launch_method'])
         launch_methods.add(self._cfg['task_launch_method'])
```

Comment:
The task_lm would also need a guard?  Or we use `cfg.get()` and later ignore the `None` in the set?